app/src/main/java/com/eis/phoneringer/RingCommands/RingtoneHandler.java

```diff
@@ -0,0 +1,103 @@
+package com.eis.phoneringer.RingCommands;
+
+import android.content.Context;
+import android.media.AudioAttributes;
+import android.media.Ringtone;
+import android.media.RingtoneManager;
+import android.net.Uri;
+import android.os.Build;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.RequiresApi;
+
+/**
+ * Class that interfaces with the RingtoneManager class of the Android library
+ *
+ * @author Alessandra Tonin, Luca Crema, Alberto Ursino
+ */
+public class RingtoneHandler {
+
+    private static Ringtone defaultRingtone;
+    private static RingtoneHandler ringtoneHandler = new RingtoneHandler();
```

Comment:
Shouldn't you rename this to `instance`? I think that's the default for Singleton pattern

Reply:
Also now that getInstance does the job of instantiating the class, you should remove `new RingtoneHandler()`